app/src/app/components/contents/download/DownloadButton.tsx

```diff
@@ -0,0 +1,39 @@
+import {FileMetadata, RootState} from "../../../../types";
+import {useState} from "react";
+import appConfig from "../../../../config/appConfig";
+import {download} from "../../../../lib/download";
+import {Button} from "../../Base/Button";
+import {DownloadCloud} from "lucide-react";
+import {bytesToSize} from "../../../../helpers";
+import {useSelector} from "react-redux";
+
+interface DownloadButtonProps {
+    file: FileMetadata
+}
+
+export default function DownloadButton({ file }: DownloadButtonProps) {
+    const [error, setError] = useState("");
+    const { isAuthenticated } = useSelector((state: RootState) => state.login );
```

Comment:
pretty sure you dont have check authenicated here.... as you have to be authenticated to see this page